07-complex-data-type/src/ComplexDataTypeSample.java

```diff
@@ -0,0 +1,258 @@
+
+public class ComplexDataTypeSample {
+
+    public static void main(String[] args) throws ComplexDataParseException {
+
+        /**
+         * Creating Mutable StoreList and StoreMap using newList() and newMap() methods
+         */
+
+        StoreList list = StoreList.newList()
+                .with("john").with('R')
+                .with(10)
+                .with(true)
+                .with(100101010L)
+                .with(98.6d)
+                .with(new byte[]{1, 2, 3, 4});
+
+        StoreMap map = StoreMap.newMap()
+                .with("name", "john")
+                .with("initial", 'R')
+                .with("age", 10)
+                .with("alive", true)
+                .with("marker", 100101010L)
+                .with("temp", 98.6d)
+                .with("img", new byte[]{1, 2, 3, 4});
+
+        System.out.println("StoreList is : " + list);
+        System.out.println("StoreMap is : " + map);
+        System.out.println();
+
+        /**
+         * Create a StoreMap and StoreList by parsing the String.
+         */
+
+        StoreMap mapUsingStringMapOF = StoreMap.mapOf("{ 'A' : " + 1 + ", 'F' : " + 1 + "}");
+        StoreList listUsingListOf = StoreList.listOf("[" + 1 + ", 'A'" + "]");
+
+        System.out.println("StoreList using listOf method : " + listUsingListOf);
+        System.out.println("StoreMap using mapOf method : " + mapUsingStringMapOF);
+        System.out.println();
+
+        /**
+         * Create a map from a {@link CellSet}.
+         * Names of cells become keys, type/value pairs become values.
+         */
+
+        StoreMap mapUsingCellMapOf = StoreMap.mapOf(originalCellSet());
+        System.out.println("Map using mapOf(Cellset) method : " + mapUsingCellMapOf);
+        System.out.println();
+
+        /**
+         * A new deep-copy mutable instance can be created from any instance via the mutableCopy() method.
+         * Make deep copy of this map and list, ensuring it is mutable.
+         */
+
+        StoreList mutableList = list.mutableCopy();
+        StoreMap mutableMap = map.mutableCopy();
+
+        System.out.println("Mutable StoreList is : " + mutableList);
+        System.out.println("Mutable StoreMap is : " + mutableMap);
+
+        /**
+         * Text representation of list and map using toString() method
+         */
+
+        String listAsString = list.toString(false);
+        String mapAsString = map.toString(false);
+
+        System.out.println("StoreList in text representation : " + listAsString);
+        System.out.println("StoreMap in text representation : " + mapAsString);
+        System.out.println();
+
+        /**
+         * Construct a single StoreList and StoreMap by parser from the provided String
+         */
+
+        StoreList convertedList = StoreList.listOf(listAsString);
+        StoreMap convertedMap = StoreMap.mapOf(mapAsString);
+
+        System.out.println("StoreList parsed from String is : " + convertedList);
```

Comment:
Would it be nice to assert or comment on the equality of these parsed values and their original versions?